### ISSUE TITLE
RF04(L029) rule not triggered in `postgres` dialect

### ISSUE THREAD

**Issue by R3gardless** [sqlfluff/sqlfluff#6429]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When using the `RF04(L029)` rule in the following SQL query:
```SQL
CREATE TABLE test_table (
    type varchar(30) NOT NULL
);
```
setting dialect between `ansi` or `postgres` yields different result.

**ANSI Dialect**
```bash
$ sqlfluff lint test.sql --rules L029 --dialect ansi    
== [test.sql] FAIL                                                                                                                                                                        
L:   2 | P:   5 | RF04 | Keywords should not be used as identifiers.
                       | [references.keywords]
All Finished 📜 🎉!
```

**Postgres Dialect**
```bash
$ sqlfluff lint test.sql --rules L029 --dialect postgres
All Finished 📜 🎉!        
```

In the `ansi` and `mysql` dialects, `type` is recognized as a keyword, and the rule triggers an error.
However, in the `postgres` dialect no error is raised.

**MySQL Dialect**
```bash
sqlfluff lint test.sql --rules L029 --dialect mysql   
== [test.sql] FAIL                                                                                                                                            
L:   4 | P:   5 | RF04 | Keywords should not be used as identifiers.
                       | [references.keywords]
All Finished 📜 🎉!
```

### Expected Behaviour

`Postgres` dialect should display the same warning as shown below:
```bash
== [test.sql] FAIL                                                                                                                                                                        
L:   2 | P:   5 | RF04 | Keywords should not be used as identifiers.
                       | [references.keywords]
```

### Observed Behaviour

`Postgres` dialect didn't display any warnings
```bash
$ sqlfluff lint test.sql --rules L029 --dialect postgres
All Finished 📜 🎉!        
```

### How to reproduce

test.sql
```SQL
CREATE TABLE test_table (
    type varchar(30) NOT NULL
);
```

Run below command
```bash
$ sqlfluff lint test.sql --rules L029 --dialect postgres
```

### Dialect

postgres

### Version

3.2.5

### Configuration

None

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
